File: app/src/main/java/dcu/fossilfuel/api/ApiService.java
package dcu.fossilfuel.api;

import java.util.List;

import dcu.fossilfuel.Guestbook;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface ApiService {
    @GET("guestbook/get")
    Call<List<Guestbook>> getGuestbookEntries();

    @POST("guestbook/post")
    Call<Guestbook> addGuestbookEntry(@Body Guestbook entry);
}

File: app/src/main/java/dcu/fossilfuel/data/api/ApiService.kt
package dcu.fossilfuel.data.api

import dcu.fossilfuel.data.model.Guestbook
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface ApiService {
    @GET("guestbook/get")
    fun getGuestbookEntries(): Call<List<Guestbook>>

    @POST("guestbook/post")
    fun addGuestbookEntry(@Body entry: Guestbook): Call<Guestbook>
}